2140 do not delete good accounts

mlissner opened the pull request:
This should be a fairly simple one:

 - Fixes a bug on the account deletion page where we weren't building a URL correctly for people that had tags
 - Makes it so we don't delete any account that ever logged in
 - Simplifies the account deletion code into a function
 - Sets stub accounts to `is_active=False`, something that shouldn't change anything but seems like a best practice

albertisfu commented on cl/users/management/commands/cl_account_management.py:
```diff
@@ -57,29 +57,31 @@ def handle(self, *args, **options):
+            if self.options["simulate"]:
+                return
+
+            delete_user_assets(up.user)
```
Is it necessary to call this method here? since we'll delete only accounts that never logged in. Is it not supposed that these kinds of accounts wouldn't have any of that kind of data?

mlissner replied:
Good call. Removed.

albertisfu commented on cl/users/templates/profile/delete.html:
```diff
@@ -16,7 +16,7 @@ <h3>This <em>cannot</em> be undone.</h3>
       possible.
   </p>
-  {% if request.user.alerts.count or request.user.favorites.count or non_deleted_map_count %}
+  {% if request.user.alerts.count or request.user.favorites.count or non_deleted_map_count or request.user.user_tags.count%}
```
```suggestion
  {% if request.user.alerts.count or request.user.favorites.count or non_deleted_map_count or request.user.user_tags.count or request.user.docket_alerts.count%}
```

Since the method `delete_user_assets` also deletes DocketAlerts, would be useful to add it here?

mlissner replied:
Yep. Good point.

albertisfu commented on cl/users/templates/profile/delete.html:
```diff
@@ -29,7 +29,7 @@ <h3>This <em>cannot</em> be undone.</h3>
     {% endif %}
     {% if request.user.user_tags.count > 0 %}
-      <li>All of your <a href="{% url "tag_list" %}">tags</a> ({{ request.user.user_tags.count }}).</li>
+      <li>All of your <a href="{% url "tag_list" username=user.username %}">tags</a> ({{ request.user.user_tags.count }}).</li>
```
```suggestion
      <li>All of your <a href="{% url "tag_list" username=user.username %}">tags</a> ({{ request.user.user_tags.count }}).</li>
{% endif %}
{% if request.user.docket_alerts.count > 0 %}
      <li>All of your <a href="{% url "profile_alerts" %}">Docket Alerts</a> ({{ request.user.docket_alerts.count }}).</li>
```

According to the previous comment.  If it's useful to show how many docket alerts will be deleted, we should add them here.

albertisfu commented:
Seems good to me! I tested the command and the Delete page and some questions emerged. I wrote them above.

albertisfu reviewed (commented):
Seems good to me! I tested the command and the Delete page and some questions emerged. I wrote them above.

mlissner commented:
@albertisfu, thanks for the review. I've made each of your fixes. 

I also added a bit of code to fix #1399, which calls for a password prompt before people delete their account. All of the work for that is in [dc53f5e](https://github.com/freelawproject/courtlistener/pull/2141/commits/dc53f5e7441b6db085fc97af02d82f405b033fa8). This fix is mostly straightforward, but it does make one tweak that concerns you: I removed the link to cancel the deletion. My thought was that it simplified the form, and that you could add your new tabbed navigation to this page once this is merged, so the cancel link wouldn't be as necessary.

I also fixed a couple small autofocus bugs that I noticed. 

Would you mind taking a second look please?

albertisfu commented on cl/users/utils.py:
```diff
@@ -61,6 +61,7 @@ def convert_to_stub_account(user: User) -> User:
     """
     user.first_name = "Deleted"
     user.last_name = ""
+    user.is_active = False
```
I think I caught a bug related to this change, maybe it's not common but I believe it's possible to happen to some users that follow the next actions:

- User delete their account from the delete page
- The same user decides to register again with the same email address in the future

The user won't be able to log in since their account was set `is_active` to `False` and this is not overridden when registering again.

Also, the user won't need to confirm their email address again because it was previously verified, might be necessary to set `profile.email_confirmed` to `False` when stubbing the account?

mlissner replied:
I don't think this should be an issue. If there's an existing, inactive, stub account, it won't be used when somebody signs up with an account. Instead, they'll just create a new one. We allow more than one account with a single email address! Django is weird sometimes!
